bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,622 @@
+  }
+
+  private void setValueInternal(long columnId, long value) {
+    for (int i = 0; i < this.bitCount(); i += 1) {
+      if ((value & (1L << i)) > 0) {
+        this.bA[i].add(columnId);
+      } else {
+        this.bA[i].removeLong(columnId);
+      }
+    }
+    this.ebM.add(columnId);
+  }
+
+  private void ensureCapacityInternal(long minValue, long maxValue) {
+    if (ebM.isEmpty()) {
+      this.minValue = minValue;
+      this.maxValue = maxValue;
+      grow(Long.toBinaryString(maxValue).length());
+    } else if (this.minValue > minValue) {
+      this.minValue = minValue;
+    } else if (this.maxValue < maxValue) {
+      this.maxValue = maxValue;
+      grow(Long.toBinaryString(maxValue).length());
+    }
+  }
+
+  private void grow(int newBitDepth) {
+    int oldBitDepth = this.bA.length;
+
+    if (oldBitDepth >= newBitDepth) {
+      return;
+    }
+
+    Roaring64NavigableMap[] newBA = new Roaring64NavigableMap[newBitDepth];
+    if (oldBitDepth != 0) {
+      System.arraycopy(this.bA, 0, newBA, 0, oldBitDepth);
+    }
+
+    for (int i = newBitDepth - 1; i >= oldBitDepth; i--) {
+      newBA[i] = new Roaring64NavigableMap();
+      if (this.runOptimized) {
+        newBA[i].runOptimize();
+      }
+    }
+    this.bA = newBA;
+  }
+
+  public void setValues(List<Pair<Long, Long>> values) {
+    if (values == null || values.isEmpty()) return;
+    long maxValue = values.stream().mapToLong(Pair::getRight).filter(Objects::nonNull).max().getAsLong();
+    long minValue = values.stream().mapToLong(Pair::getRight).filter(Objects::nonNull).min().getAsLong();
+    ensureCapacityInternal(minValue, maxValue);
+    for (Pair<Long, Long> pair : values) {
+      setValueInternal(pair.getKey(), pair.getValue());
+    }
+  }
+
+  /**
+   * merge will merge 2 bsi into current
+   * merge API was designed for distributed computing
+   * note: current and other bsi has no intersection
+   *
+   * @param otherBsi other bsi we need merge
+   */
+  public void merge(Roaring64NavigableMapSliceIndex otherBsi) {
+
+    if (null == otherBsi || otherBsi.ebM.isEmpty()) {
+      return;
+    }
+
+    // todo whether we need this
+    if (Roaring64NavigableMap.intersects(this.ebM, otherBsi.ebM)) {
+      throw new IllegalArgumentException("merge can be used only in bsiA  bsiB  is null");
+    }
+
+    int bitDepth = Integer.max(this.bitCount(), otherBsi.bitCount());
+    Roaring64NavigableMap[] newBA = new Roaring64NavigableMap[bitDepth];
+    for (int i = 0; i < bitDepth; i++) {
+      Roaring64NavigableMap current = i < this.bA.length ? this.bA[i] : new Roaring64NavigableMap();
+      Roaring64NavigableMap other = i < otherBsi.bA.length ? otherBsi.bA[i] : new Roaring64NavigableMap();
+      newBA[i] = Roaring64NavigableMap.or(current, other);
+      if (this.runOptimized || otherBsi.runOptimized) {
+        newBA[i].runOptimize();
+      }
+      newBA[i].setSerializationMode(Roaring64NavigableMap.SERIALIZATION_MODE_PORTABLE);
```

Comment:
Changing the serialization mode here looks weird, especially as you may mutate input bitmaps. Also, be aware you re toggling a static field field: it impacts the whole JVM (hence it is a sensitive call, and unecessary to be done on a per-bitmap basis)

(@lemire I feel we missed the opportunity to change the default to portable format with 1.X:
```
  // As of RoaringBitmap 0.X, we stick to the legacy format for retrocompatibility
  // RoaringBitmap 1.X may switch to the portable format by default
  public static int SERIALIZATION_MODE = SERIALIZATION_MODE_LEGACY;
```

)

Reply:
got it, would it be better to set the 'SERIALIZATION_MODE' field as a local variable?

And that's the place where i also feel so weird. if we don't consider too much backward compatibility, maybe default to portable format?

Reply:
what about set the 'SERIALIZATION_MODE' as a local variable(rename as serializationMode), and set the default value to SERIALIZATION_MODE_PORTABLE. And add a setter method for those who wants to switch to the legacy format. @lemire @blacelle